extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -538,6 +567,179 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
+    const init = {
+      credentials: 'include',
+    };
+    // If stale-while-revalidate is disabled, our first RTC callout
+    // needs to bypass cache, so include the header that will
+    // bypass.
+    if (disableSWR) {
+      init.headers = headers;
+    }
+    const startTime = Date.now();
+    // Because we are wrapping the RTC request in the timeout,
+    // we are guaranteeing that if the RTC is slow to return and
+    // times out for the first slot, it won't be used for any
+    // other slots either. This is in opposition to the other way
+    // we could potentially do it, in which we only still make one
+    // rtc callout, but there is a 1 second timeout from the
+    // perspective of each slot. I.e. if the top slot on the page
+    // calls out for the RTC, which returns after 5 seconds, and
+    // then a slot way down on the page asks for it later.
+    rtcPromise = Services.timerFor(window).timeoutPromise(
+        RTC_TIMEOUT,
+        Services.xhrFor(this.win).fetchJson(
+            endpoint, init).then(res => {
+              rtcTotalTime = Date.now() - startTime;
+              if (!disableSWR) {
+                // Repopulate the cache.
+                Services.xhrFor(this.win).fetchJson(endpoint, {
+                  credentials: 'include',
+                  headers,
+                }).catch(err => {
+                  user().error(TAG, err.message);
+                });
+              }
+              // Non-200 status codes are forbidden for RTC.
+              // TODO: Add to fetchResponse the ability to
+              // check for redirects as well.
+              if (res.status != 200) {
+                return {rtcTotalTime};
+              }
+              return res.text().then(text => {
+                // An empty text response is fine, just means
+                // we have nothing to merge.
+                if (!text) {
+                  return {rtcTotalTime, success: true};
+                }
+                const rtcResponse = tryParseJson(text);
+                return {rtcResponse, rtcTotalTime};
+              });
+            }));
+
+    return this.mergeRtc();
+  }
+
+  /**
+   * Merges the RTC response into the jsonTargeting of this.
+   * If it can't merge, or there is no response, potentially
+   * rejects.
+   * @return {Promise<?Object>} Resolves if ad request is
+   *     to be sent, with object of params to add to request,
+   *     otherwise rejects with a reject message if we have one.
+   */
+  mergeRtc() {
+    // add reasons for promise.reject
+    return rtcPromise.then(
+        r => {
+          // Don't try to merge if we're sending without RTC.
+          if (!r || (!r.rtcResponse && !r.success)) {
+            return this.shouldSendRequestWithoutRtc(
+                'Bad response');
+          } else if (!r.rtcResponse && r.success) {
+            return Promise.resolve({
+              artc: r.rtcTotalTime,
+              ati: 2,
```

Comment:
use your new enum here, can you also add a comment indicating that this is the case where you got an empty response so no need to merge